### DISCUSSION TITLE
Something about function mg_connect

### DISCUSSION THREAD

**Discussion by hywing** [cesanta/mongoose#2435] [Ask questions, get answers] [answered]
Can I connect a tcp url with function **mg_connect**? 
`mg_connect(&mgr, "tcp://0.0.0.0:1234", cb, &mgr);`
I got error when I tried doing this
`Connection created
1494a28 1 mongoose.c:403:mg_error       1 0xcc socket error
Connection closed
`

**Comment by hywing** [comment]
It turns out to be the right url is `tcp://localhost:1234` not `tcp://0.0.0.0:1234`

**Comment by scaprile** [comment, marked as answer]
Please follow the guidelines in the [documentation](https://mongoose.ws/documentation), [examples](https://github.com/cesanta/mongoose/tree/master/examples) and [tutorials](https://mongoose.ws/tutorials/) available.
There is already an example on how to do what you want to do: https://mongoose.ws/documentation/tutorials/tcp/
Yours is a networking concepts problem. `localhost` means "resolve this address and use it", that is usually the loopback address; `0.0.0.0` means "this host in this network" and in a listener context means "listen in all interfaces". You can't, of course, try to connect to that address because there is no way to resolve that,
